src/diffenator2/matcher.py

```diff
@@ -0,0 +1,125 @@
+"""
+Match font styles or match vf coordinates
+"""
+from diffenator2.font import get_font_styles, Style
+from fontTools.ttLib.scaleUpem import scale_upem
+import re
+
+
+class FontMatcher:
+
+    def __init__(self, old_fonts, new_fonts):
+        self.old_fonts = old_fonts
+        self.new_fonts = new_fonts
+        self.old_styles = []
+        self.new_styles = []
+    
+    def _match_fonts(self):
+        old_fonts = []
+        new_fonts = []
+        for new_font in self.new_fonts:
+            new_names = self._get_names(new_font)
+            best_font = None
+            match_count = -float("inf")
+            for old_font in self.old_fonts:
+                if old_font == new_font:
+                    continue
+                old_names = self._get_names(old_font)
+                matched_names = old_names & new_names
+                if len(matched_names) > match_count:
+                    match_count = len(matched_names)
+                    best_font = old_font
+            old_fonts.append(best_font)
+            new_fonts.append(new_font)
+        self.old_fonts = old_fonts
+        self.new_fonts = new_fonts
+
+    def _get_names(self, font):
+        results = set()
+        name = font.ttFont["name"]
+        fvar = font.ttFont["fvar"]
+        # TODO add STAT
+        results.add(name.getBestSubFamilyName())
+        for inst in fvar.instances:
+            results.add(name.getName(inst.subfamilyNameID, 3, 1, 0x409).toUnicode())
+        return results
+    
+    def diffenator(self, coords=None):
+        assert len(self.old_fonts) == 1 and len(self.new_fonts) == 1, "Multiple fonts found. Diffenator can only do a 1 on 1 comparison"
+        old_font = self.old_fonts[0]
+        new_font = self.new_fonts[0]
+        if old_font.is_variable() and new_font.is_variable():
+            if not coords:
+                coords = {a.axisTag: a.defaultValue for a in new_font.ttFont["fvar"].axes}
+            self.old_styles.append(Style(old_font, coords, ""))
+            self.new_styles.append(Style(new_font, coords, ""))
+            old_font.set_variations(coords)
+            new_font.set_variations(coords)
+        elif old_font.is_variable() and not new_font.is_variable():
+            self.instances()
+            old_font.set_variations_from_static_font(new_font)
+        elif new_font.is_variable() and not old_font.is_variable():
+            self.instances()
+            new_font.set_variations_from_static_font(old_font)
+        elif not old_font.is_variable() and not new_font.is_variable():
+            self.old_styles.append(Style(old_font, {"wght": 400}, ""))
```

Comment:
I first read this and thought "wait, what, assuming Regular weight?" before I realised what was going on. A comment would be helpful.